packages/react/src/components/Popover/Popover.tsx

```diff
@@ -46,131 +15,81 @@ export type PopoverProps = {
    * @default default
    */
   variant?: 'default' | 'info' | 'warning' | 'danger';
-  /** Whether the tooltip is open or not.
-   * This overrides the internal state of the tooltip.
+  /**
+   * Use this to make the popover controlled.
+   * @default undefined
    */
   open?: boolean;
   /** Size of the popover
    * @default small
    */
   size?: 'small' | 'medium' | 'large';
-  /** Callback function when popover closes */
+  /** Callback function when popover changes open state */
+  onOpenChange?: (open: boolean) => void;
+  /**
+   * Callback when the popover wants to close.
+   */
   onClose?: () => void;
 } & PortalProps &
   HTMLAttributes<HTMLDivElement>;
 
-export const Popover = forwardRef<HTMLDivElement, PopoverProps>(
-  (
-    {
-      children,
-      placement = 'top',
-      open,
-      anchorEl,
-      variant = 'default',
-      size = 'small',
-      onClose,
-      portal,
-      className,
-      style,
-      ...rest
-    },
-    ref,
-  ) => {
-    const Container = portal ? FloatingPortal : React.Fragment;
-    const arrowRef = useRef<HTMLDivElement>(null);
-    const floatingEl = useRef<HTMLDivElement>(null);
-
-    const {
-      context,
-      update,
-      refs,
-      floatingStyles,
-      placement: flPlacement,
-      middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
-    } = useFloating({
-      placement,
-      open,
-      onOpenChange: () => onClose && onClose(),
-      whileElementsMounted: autoUpdate,
-      elements: {
-        reference: anchorEl,
-        floating: floatingEl.current,
-      },
-      middleware: [
-        offset(ARROW_HEIGHT + ARROW_GAP),
-        flip({
-          fallbackAxisSideDirection: 'start',
-        }),
-        shift(),
-        arrow({
-          element: arrowRef,
-        }),
-      ],
-    });
-
-    const { getFloatingProps } = useInteractions([
-      useFocus(context),
-      useClick(context),
-      useDismiss(context),
-      useRole(context),
-    ]);
+export const Popover = ({
+  children,
+  placement = 'top',
+  open,
+  variant = 'default',
+  size = 'small',
+  onOpenChange,
+  onClose,
+  portal,
+}: PopoverProps) => {
+  const anchorEl = useRef<Element>(null);
```

Comment:
Was it not possible to have support alongside `anchorEl` passing it as the initial value to `useRef` here?

Reply:
Not passing it to the initial `anchorEl` I think, but I have an idea I'll test out